Buildings/Experimental/DHC/CentralPlants/Cooling/Subsystems/CoolingTowerWithBypass.mo

```diff
@@ -0,0 +1,397 @@
+  Buildings.Fluid.Actuators.Valves.TwoWayEqualPercentage valByp(
+    redeclare final package Medium=Medium,
+    final m_flow_nominal=m_flow_nominal*0.0001,
+    final dpValve_nominal=dp_nominal,
+    final use_inputFilter=use_inputFilter)
+    "Condenser water bypass valve"
+    annotation (Placement(transformation(extent={{-10,-10},{10,10}},origin={0,-40})));
+  Buildings.Fluid.Sensors.TemperatureTwoPort senTCWSup(
+    redeclare final package Medium=Medium,
+    final m_flow_nominal=m_flow_nominal,
+    final T_start=Medium.T_default)
+    "Temperature sensor"
+    annotation (Placement(transformation(extent={{60,10},{80,-10}})));
+  Modelica.Blocks.Sources.Constant TSetByPas(
+    final k=TMin)
+    "Bypass loop temperature setpoint"
+    annotation (Placement(transformation(extent={{-90,-60},{-70,-40}})));
+  Buildings.Controls.Continuous.LimPID bypValCon(
+    u_s(
+      final unit="K",
+      displayUnit="degC"),
+    u_m(
+      final unit="K",
+      displayUnit="degC"),
+    controllerType=Modelica.Blocks.Types.SimpleController.PI,
+    k=1,
+    Ti=60,
+    final reset=Buildings.Types.Reset.Parameter,
+    final y_reset=0)
+    "Bypass valve controller"
+    annotation (Placement(transformation(extent={{-60,-60},{-40,-40}})));
+  Buildings.Controls.Continuous.LimPID cooTowSpeCon(
+    u_s(
+      final unit="K",
+      displayUnit="degC"),
+    u_m(
+      final unit="K",
+      displayUnit="degC"),
+    final reverseActing=false,
+    final controllerType=controllerType,
+    final k=k,
+    final Ti=Ti)
+    "Cooling tower fan speed controller"
+    annotation (Placement(transformation(extent={{-12,50},{8,70}})));
+  Buildings.Controls.OBC.CDL.Logical.Switch swi
+    "Output the input of higher value"
+    annotation (Placement(transformation(extent={{-40,50},{-20,70}})));
+  Buildings.Controls.OBC.CDL.Continuous.Greater gre
+    "Compare if (TWetBul+dTApp) is greater than TMin"
+    annotation (Placement(transformation(extent={{-70,50},{-50,70}})));
+  Buildings.Controls.OBC.CDL.Continuous.AddParameter addPar(
+    p=dTApp,
+    k=1)
+    "Add approach temperature on top of wetbulb temperature"
+    annotation (Placement(transformation(extent={{-80,80},{-60,100}})));
+equation
+  connect(cooTowSys.TWetBul,TWetBul)
+    annotation (Line(points={{-12,-6},{-40,-6},{-40,-20},{-120,-20}},color={0,0,127}));
+  connect(port_a,cooTowSys.port_a)
+    annotation (Line(points={{-100,0},{-10,0}},color={0,127,255}));
+  connect(cooTowSys.port_b,senTCWSup.port_a)
+    annotation (Line(points={{10,0},{60,0}},color={0,127,255}));
+  connect(senTCWSup.port_b,port_b)
+    annotation (Line(points={{80,0},{100,0}},color={0,127,255}));
+  connect(TSetByPas.y,bypValCon.u_s)
+    annotation (Line(points={{-69,-50},{-62,-50}},color={0,0,127}));
+  connect(senTCWSup.T,bypValCon.u_m)
+    annotation (Line(points={{70,-11},{70,-80},{-50,-80},{-50,-62}},color={0,0,127}));
+  connect(valByp.port_a,cooTowSys.port_a)
```

Comment:
I believe the ideal mixing at `valByp.port_a` should have a junction. Because the `port_a` of this model can be connected externally, it is possible (and likely) that ideal mixing at this location creates a false mixing point that doesn't occur in the real system. Conversely, because of the temperature sensor at the downstream `valByp.port_b`, I think ideal mixing is okay in this second case. 
Could you please add a `Buildings.Fluid.FixedResistances.Junction` at the upstream mixing point of the `cooTowSys` and `valByp`? This junction can have `dp_nominal={0,0,0}`, and `energyDynamics=SteadyState`

See https://simulationresearch.lbl.gov/modelica/userGuide/bestPractice.html#modeling-of-fluid-junctions for more details.